src/markitdown/_markitdown.py

```diff
@@ -676,19 +677,27 @@ def convert(self, local_path, **kwargs) -> Union[None, DocumentConverterResult]:
 
 class PdfConverter(DocumentConverter):
     """
-    Converts PDFs to Markdown. Most style information is ignored, so the results are essentially plain-text.
+    Converts PDFs to Markdown. Most style information is ignored, so the results are essentially plain-text.    
     """
 
-    def convert(self, local_path, **kwargs) -> Union[None, DocumentConverterResult]:
+    def convert(self, local_path, pdf_engine: Literal['pdfminer', 'pymupdf4llm']='pdfminer', **kwargs) -> Union[None, DocumentConverterResult]:
+        """
+        Example:
+        >>> source = "https://arxiv.org/pdf/2308.08155v2.pdf"
+        >>> markitdown.convert(source, pdf_engine="pymupdf4llm")
+        """
         # Bail if not a PDF
         extension = kwargs.get("file_extension", "")
         if extension.lower() != ".pdf":
             return None
-
-        return DocumentConverterResult(
-            title=None,
-            text_content=pdfminer.high_level.extract_text(local_path),
-        )
+        if pdf_engine == "pdfminer":
+            text_content = pdfminer.high_level.extract_text(local_path)
+        elif pdf_engine == "pymupdf4llm":
+            text_content = pymupdf4llm.to_markdown(local_path, show_progress=False)
+        else:
+            # return None     # unknown method
+            raise FileConversionException("'pdf_engine' not valid. Please choose between ['pdfminer', 'pymupdf4llm'].")
```

Comment:
I'd suggest checking the engine first. You could use `_engines` to define the allowed engines

```py
_engines: Mapping[str, Any] = {
        "pdfminer": pdfminer,
        "pymupdf4llm": pymupdf4llm,
    }
    
###
if engine is not None and engine not in self._engines:
    raise
```

Reply:
It is a good idea for the ease of adding further engines. I have updated those changes in the latest commit. An example of using `engine_kwargs` for pdf imeges extraction could also be found in the test file.